### PR TITLE
refactor(input): migrate to signals

### LINKED ISSUE #421: refactor(input): migration to signals
### Which scope/s are relevant/related to the feature request?

input

### Information

Migrate away from all Input/Output and View Decorators

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by marcjulian**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] icon
- [x] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

Closes #421

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/ui/form-field/helm/src/lib/hlm-form-field.component.ts [hunk 1]
```diff
@@ -23,13 +22,13 @@ import { HlmErrorDirective } from './hlm-error.directive';
 	},
 })
 export class HlmFormFieldComponent {
-	control = contentChild(BrnFormFieldControl);
+	private readonly control = contentChild(BrnFormFieldControl);
```
Comment (goetzrobin) [inline libs/ui/form-field/helm/src/lib/hlm-form-field.component.ts:25]: should these be private? I feel like people might want to access them programatically?

Comment (marcjulian) [reply libs/ui/form-field/helm/src/lib/hlm-form-field.component.ts:25]: yes probably better to change to public readonly

**Review by goetzrobin — APPROVED** [review]

**Comment by goetzrobin** [comment]
@marcjulian the chromatic build failed with a type error for storybook. can you check that?

File: libs/ui/input/helm/src/lib/hlm-input.directive.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ type InputVariants = VariantProps<typeof inputVariants>;
 export class HlmInputDirective implements BrnFormFieldControl, DoCheck {
 	public readonly size = input<InputVariants['size']>('default');
 
-	public readonly error = signal<InputVariants['error']>('auto');
+	public readonly error = input<InputVariants['error']>('auto');
```
Comment (marcjulian) [inline libs/ui/input/helm/src/lib/hlm-input.directive.ts:48]: @goetzrobin should be fixed now. Somehow I forgot to change error to `input` which caused the type error in storybook.

**Pull request merged**